excerise.go

```diff
@@ -0,0 +1,61 @@
+package main
+import "fmt"
+func worker(id int, jobs <-chan int, results chan<- int) {
+						
+for j := range jobs { go func() {
+						
+switch j % 3 { case 0:
+						
+j=j*1 case 1:
+						
+j=j*2
+						
+results <- j * 2 case 2:
+						
+results <- j * 3
+						
+j=j*3 }
+						
+}() }
+						
+}
+func main() {
+						
+jobs := make(chan int)
+results := make(chan int)
+for i := 1; i <= 1000000000; i++ {
+						
+go func() {
+if i%2 == 0 {
+						
+i += 99 }
+						
+jobs <- i }()
+						
+}
+close(jobs)
```

Comment:
We have just started a billion of goroutines which send data to `jobs` channel but now we are closing the channel without any synchronisation and checking if all goroutines have finished. Also the `jobs` channel is not used for anything meaningful. 
Perhaps the fix can be to `defer close(jobs)`